hook.php

```diff
@@ -503,25 +477,16 @@ function plugin_formcreator_hook_delete_ticket(CommonDBTM $item) {
 
 function plugin_formcreator_hook_restore_ticket(CommonDBTM $item) {
    $formAnswer = new PluginFormcreatorFormAnswer();
-   $formAnswer->getFromDBByCrit([
-      'id' => new QuerySubQuery([
-         'SELECT' => 'items_id',
-         'FROM'   => Item_Ticket::getTable(),
-         'WHERE'  => [
-            'itemtype' => PluginFormcreatorFormAnswer::getType(),
-            'tickets_id' => $item->getID(),
-         ]
-      ])
-   ]);
-   if ($formAnswer->isNewItem()) {
-      plugin_formcreator_hook_add_ticket($item);
+   if ($formAnswer->getFromDbByTicket($item)) {
+      $formAnswer->createIssue();
+      $minimalStatus = $formAnswer->getAggregatedStatus();
+      if ($minimalStatus !== null) {
+         $formAnswer->updateStatus($minimalStatus);
+      }
       return;
    }
```

Comment:
if statement inverted

when there is a form answer associated to the ticket then the status of the form answer must be updated by the ticket restoratin event because several tickets may be ssociated to this form answer instance.

If there is no form answer associated to the ticket, then a new issue must be recreated from the restored ticket.